### ISSUE TITLE
Api parity with pytorch for optimizers/ model/ schedulers et al

### ISSUE THREAD

**Issue by robert-wagner** [OpenMined/PySyft#2213]
**Describe the solution you'd like**
Pytorch specifies a specific api for training models. Specifically this api includes defining a model, initializing an optimizer in relation to that model then initializing a scheduler to schedule optimization steps.

We should make sure that the api of PySyft preserves as much of this api as possible to ease users transitioning from pure Pytorch.

**Comment by LaRiffle** [comment]
Do you have any resource detailing this API that you could share?
